src/main/diagnostic.ts

```diff
@@ -0,0 +1,86 @@
+import type { PortMapping } from "@common/types";
+import { NatType, Presence } from "@common/types";
+import { createPmpClient, createUpnpClient } from "@xmcl/nat-api";
+import { gateway4async } from "default-gateway";
+import Tracer from "nodejs-traceroute";
+import { createServer, request } from "stun";
+
+export async function getNatType(): Promise<{ address: string; natType: NatType }> {
```

Comment:
If these functions are always called together we don't need to export all three functions. Just export a single function `getNetworkDiagnostics()` which calls all the other functions and returns everything that we need to render in the FE.